crypto/dsa/ecdsa/ecdsa.go

```diff
@@ -0,0 +1,83 @@
+package ecdsa
+
+import (
+	"fmt"
+
+	"github.com/tbd54566975/web5-go/jwk"
+)
+
+const (
+	KeyType = "EC"
+)
+
+var AlgorithmIDs = map[string]bool{
+	SECP256K1AlgorithmID: true,
+}
+
+func GeneratePrivateKey(algorithmID string) (jwk.JWK, error) {
+	var privateKey jwk.JWK
+	var err error
+
+	switch algorithmID {
+	case SECP256K1AlgorithmID:
+		privateKey, err = SECP256K1GeneratePrivateKey()
+	default:
+		err = fmt.Errorf("unsupported algorithm: %s", algorithmID)
+	}
+
+	return privateKey, err
+}
+
+func GetPublicKey(privateKey jwk.JWK) jwk.JWK {
+	return jwk.JWK{
+		KTY: privateKey.KTY,
+		CRV: privateKey.CRV,
+		X:   privateKey.X,
+		Y:   privateKey.Y,
+	}
+}
+
+func Sign(payload []byte, privateKey jwk.JWK) ([]byte, error) {
+	if privateKey.D == "" {
+		return nil, fmt.Errorf("d must be set")
+	}
+
+	var signature []byte
+	var err error
+
+	switch privateKey.CRV {
+	case SECP256K1JWACurve:
+		signature, err = SECP256K1Sign(payload, privateKey)
+	default:
+		err = fmt.Errorf("unsupported curve: %s", privateKey.CRV)
+	}
+
+	return signature, err
+}
+
+func Verify(payload []byte, signature []byte, publicKey jwk.JWK) (bool, error) {
+	var valid bool
```

Comment:
As above, just return directly from the switch rather than falling through with the values. The latter is quite a bit more error prone too, because you could forget to populate the values and then the function might return garbage.

Reply:
done: https://github.com/TBD54566975/web5-go/commit/44797d08c88d0de7ae45d4dae8508fc2d6b23776